PythonPonies/bangazon-storefront
# Customer Can Add/View Payment Types

# Description
This pull request covers the application requirement to afford the user the ability to view their existing payment types, and to add new payment types to associate with their account. 

## Number of Fixes
0 fixes. 

## Related Ticket(s)
#3

## Problem to Solve
I wanted the application to filter all existing payment types by user id, and return the current user's payment types. In addition, I wanted the user to be able to add new payment type information, and upon adding, the new payment type to be displayed with other existing payment types. 

## Proposed Changes
I refactored the PaymentType view to enable appropriate database querying. 

## Expected Behavior
The user should now be able to log in, and view their existing payment types, along with add new payment types to associate with their account. 

## Steps to Test Solution

1. git fetch
2. git checkout swh-completeorder [disregard name - i made in error when naming branch]
3. rm db.sqlite3
4. python manage.py makemigrations
5. python manage.py migrate
6. python manage.py runserver
7. direct browser to localhost:8000/register
8. register new user.
9. direct browser to localhost:8000/payment
10. add new payment types to your hearts content.
11. you can also register another user and add payment types for them and see in the database that the different payment types will be associated with the correct user/customer's id. 

## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[x] I certify that all existing tests pass [with exception for the ones that are supposed to fail]

## Documentation
[ ] There is new documentation in this pull request that must be reviewed..
[ ] I added documentation for any new classes/methods

## Deploy Notes
N/A
